### ISSUE TITLE
Time stamps: time 1 and time 2

### ISSUE THREAD

**Issue by gtrsmith** [EyeSeeTea/SurveillanceCambodiaApp#49]
KH would like to see two time stamps included in analytics:

Time 1: time when a user presses + symbol to start a new survey

Time 2: time when the survey is sent to DHIS

**Comment by QISPSK** [comment]
Both DEs created in the development server and are correctly registering the date, but the time is listed as 00:00:00 for all events. Doesn't appear to be correctly capturing/displaying the time.

**Comment by QISPSK** [comment]
These DEs seem to crash Event Reports. When they are removed, Event Reports seems to work fine (in most cases). Any idea why this may be?

**Comment by ifoche** [comment]
The only special thing that these DEs have is they are Date&Time type. So...maybe DHIS2 have a bug with Date&Time types in Event Reports?

**Comment by rodmelia** [comment]
There are 2 issues here. Let's separate them:
1. The data element is expected to save the date AND TIME. IF that's not happening, the problem could be the app, the SDK or the API. Let's focus on this one - we want to capture this data correctly soon (still, not a critical bug)
2. Event reports crashing is a separate issue, and we need to report and work with Oslo to fix it. This is likely to take weeks.

**Comment by ifoche** [comment]
I agree. In principle, task 1 should be done. It's registering date & time. Do you want me to report 2?

**Comment by rodmelia** [comment]
No - just make sure that the date & TIME is properly saved in DHIS for the
control data elements.

**Comment by ifoche** [comment]
Can anyone confirm this is done?

**Comment by QISPSK** [comment]
Hi @ifoche, it seems like this issue is still a problem. While Event Reports is no longer crashing when including these DEs, both DEs are showing up without any data.

**Comment by ifoche** [comment]
Hi @QISPSK ,

I've just looked at the server and I can see them...

![time12_bug](https://cloud.githubusercontent.com/assets/10931645/16727126/dcde1118-475f-11e6-8f64-2799611d236f.png)

isn't it what you were expecting?

**Comment by QISPSK** [comment]
Strangely, no. Both DEs show up blank for all the surveys sent so far. Maybe it is just my permissions? I can have staff check here tomorrow.

![image](https://cloud.githubusercontent.com/assets/17549893/16727896/8e2288aa-478d-11e6-9f9f-b005c9c93cb1.png)

**Comment by ifoche** [comment]
Can you make a test by sending to QIS_test_int and checking that you cannot see it? BTW, which tool/webapp are you using to see it? my screenshot was from an Event Capture execution.

**Comment by rodmelia** [comment]
I can confirm that the event report is not displaying DE of type date
correctly. The information is being stored correctly in DHIS, as per
ignacio's screenshot (which shows it on the event capture).

We need to get UiO to fix that problem, but it could easily take 3-4 weeks
- they are currently closed for summer holidays.

**Comment by QISPSK** [comment]
Thanks for the update Rodolfo! It's not crucial that we have this immediately and I've downgraded it's importance to low. 3-4 weeks shouldn't be a problem.

**Comment by QISPSK** [comment]
Any updates on this? UNOPS would like to change one of our indicator to track number of cases submitted within 24 hours and we would like to use these two DEs to track this, if possible.

**Comment by josemp10** [comment]
hi, this is a dhis2 bug that has been already reported to Oslo. No answer
yet (likely because we are in 2.22 and they are about to finish with 2.25).
I will try again, but maybe this is not going to be possible until we
migrate to 2.25, likely in Jan 2017. Normally Oslo does not support that
old dhis2 versions. Will try again anyway..

**Comment by QISPSK** [comment]
Moved this back to "Ready" after sending some data to old-staging.psi-mis.org. It looks like this DE still isn't showing up in Event Reports. Any updates from Oslo on this? The hope was that this would be fixed in v2.25.

**Comment by rodmelia** [comment]
Hi Cambodia - (not sure who is writing).

Is there a hyperlink that you can send for opening this in the report view?
The time may not show on the event report, but have you tried downloading
it in excel? Generally it shows.

Also, if you are seeking to count the number of cases, perhaps you should
try doing an indicator, and call this via pivot table. Let me know.

R

**Comment by QISPSK** [comment]
@rodmelia Sorry, this is Eli. A link is below. The table sorts only for data sent from phones with S6. 

https://old-staging.psi-mis.org/dhis-web-event-reports/index.html?id=kQVIuM8CWWP

As you can see, the "KH_MAL_SRV - Time Capture" and "KH_MAL_SRV - Time Sent" DEs are blank. Downloading this table to Excel shows the Event Date (which is when it hits DHIS2, I assume?), however it does not show the corresponding time capture date. 

This data is, however, captured in the server as it can be seen in Event Capture and @josemp10 has been able to download all events for us in the past so we could do a comparison of time sent vs time capture. Going forward, it would be nice if we could do this analysis on our own without requiring @josemp10 to download the data for us.

**Comment by josemp10** [comment]
Hi! We have reported this to Oslo, but it seems like we have to do it
again. Date DateElements should be rendered in the analytics.

**Comment by rodmelia** [comment]
Jose - do you know the Jira reference?
If there isn't a Jira issue, we should send to Sam, who is managing the
error reporting.
